### ISSUE TITLE
CurrentProcessedLines are always 0

### ISSUE THREAD

**Issue by landauermax** [ait-aecid/logdata-anomaly-miner#644]
The changes made in PR #610 cause that the CurrentProcessedLines are always 0. The count dict is set to zero
`self.count_dict[k][current_processed_lines_str] = 0`
but it is later forwarded to the output through `event_data = {'StatusInfo': self.count_dict ...`. Please undo the changes of the PR or create a fix.

Sample aminer output, where the TotalProcessingLines increase from 13 to 14, but the CurrentProcessedLines are always 0 (should be 1 in the second report):

```
Count report (1 lines)
  {
  "StatusInfo": {
    "other": {
      "CurrentProcessedLines": 0,
      "TotalProcessedLines": 13
    }
  },
  "FromTime": 1618982877.511722,
  "ToTime": 1618982882.511734
}

Count report (1 lines)
  {
  "StatusInfo": {
    "other": {
      "CurrentProcessedLines": 0,
      "TotalProcessedLines": 14
    }
  },
  "FromTime": 1618982883.488207,
  "ToTime": 1618982888.488224
}
```

**Comment by dITySoftware** [comment]
Unfortunately I can not reproduce this behavior. I have tried using the aminer demo with a ParserCount with no target_path_list set and with target_path_list set to ["/model/Random", "/model/RandomTime"]. However everything seems normal to me. Could you please view the attached files and check if the outputs are valid?

[demo-output-with-target-paths.txt](https://github.com/ait-aecid/logdata-anomaly-miner/files/6349405/demo-output-with-target-paths.txt)
[demo-output-no-target-paths.txt](https://github.com/ait-aecid/logdata-anomaly-miner/files/6349408/demo-output-no-target-paths.txt)

**Comment by landauermax** [comment]
Yes, the outputs look valid. I will try to find out what could be different in my setup and get back to you with a reproducible configuration if the problem persists.

**Comment by landauermax** [comment]
Could you please try to follow my steps and see if you can reproduce it? I do not understand why the aminer demo does not show this behavior, but I tried it on two different systems and I always had the current counts set to 0.

I installed the current aminer from the development branch (it is important that the development is used, since the PR I mentioned above is not in the main branch yet) using the aminer_install.sh script as outlined in the Getting Started Wiki.

I use the following input file (test.log):

```
a
a

```

I use the following config:

```
LearnMode: True
AminerUser: 'aminer'
AminerGroup: 'aminer'

LogResourceList:
        - 'file:///home/ubuntu/tmp6/test.log'

Parser:
        - id: parser
          type: FixedDataModelElement
          name: 'parser'
          args: 'a'

        - id: model
          start: true
          type: SequenceModelElement
          name: 'model'
          args:
             - parser

Input:
        verbose: True
        multi_source: False
        timestamp_paths:
          - '/irrelevant'

Analysis:
        - type: 'ParserCount'
          paths:
            - '/model/parser'
          report_interval: 5

EventHandlers:
        - id: "stpe"
          json: true
          type: "StreamPrinterEventHandler"
```

When running the aminer, I would expect that both current and total line count is 2. But only the total line count is 2, the current count is always 0:

```
root@user-5:/home/ubuntu/tmp6# aminer -C -c config.yml 
2021-04-21 10:13:07 New path(es) detected
NewMatchPathDetector: "DefaultNewMatchPathDetector" (1 lines)
  {
  "AnalysisComponent": {
    "AnalysisComponentIdentifier": 1,
    "AnalysisComponentType": "NewMatchPathDetector",
    "AnalysisComponentName": "DefaultNewMatchPathDetector",
    "Message": "New path(es) detected",
    "PersistenceFileName": "Default",
    "AffectedLogAtomPaths": [
      "/model",
      "/model/parser"
    ],
    "ParsedLogAtom": {
      "/model": "a",
      "/model/parser": "a"
    }
  },
  "LogData": {
    "RawLogData": [
      "a"
    ],
    "Timestamps": [
      1618999987.01
    ],
    "DetectionTimestamp": 1618999987.01,
    "LogLinesCount": 1,
    "AnnotatedMatchElement": "/model: a\n  /model/parser: a"
  }
}

Count report (1 lines)
  {
  "StatusInfo": {
    "/model/parser": {
      "CurrentProcessedLines": 0,
      "TotalProcessedLines": 2
    }
  },
  "FromTime": 1618999987.021336,
  "ToTime": 1618999992.021361
}
```

**Comment by dITySoftware** [comment]
I have figured the error out and fixed it in #645. I know it should be tested in unittests, however I want to reimplement these tests at a later point.

Could you please recheck if the error still exists in that Pull Request?

**Comment by landauermax** [comment]
Thanks, yes, the PR solves the issue.
